### DISCUSSION TITLE
강의 선수지식

### DISCUSSION THREAD

**Discussion by ggobsarikuna** [tedilabs/fastcampus-devops#42] [General]
안녕하세요. 올해부터 DevOps 관련 공부를 처음 시작하게 된 학생입니다. 프로그래밍 언어, 리눅스 같은 공부도 처음 시작하고 있는 상황입니다. Part1. CH03_06 SSH Key 권한 변경 부분에서부터 막혔는데 다른 분이 올리신 질문도 참고하고 구글링을 통해서 해결하려고 해도 감이 잘 안잡힙니다. 이 강의를 이해하면서 넘어가기 위한 선수지식이 필요한지 여쭤보고 싶습니다. 번거로우시겠지만 조언 부탁드립니다.

**Comment by ggobsarikuna** [comment]
![12](https://user-images.githubusercontent.com/88932224/149913583-d2d690c7-d95b-458f-9519-d5d9d0cb39fa.PNG)
추가적으로 pastcampus.pem 권한설정 바꾸고 나서도 이런 오류 뜨는데 해결방법 부탁드립니다.

**Reply by posquit0** [reply]
안녕하세요,
강의를 잘 따라오기 위해서는 리눅스 운영체제의 기본적인 사용방법 (기본적인 리눅스 명령어 사용법, 패키지 관리 방법)에 대한 이해가 필요합니다.

https://velog.io/@devyang97/Linux-%EB%AA%85%EB%A0%B9%EC%96%B4-%EC%A0%95%EB%A6%AC-Ubuntu-%EC%82%AC%EC%9A%A9

위와 같은 글이나 관련 도서 등을 통해 리눅스 명령어와 패키지 관리에 대해 학습해보시면 좋을 듯 합니다. :)

**Reply by posquit0** [reply]
질문 주신 내용에 대해서는 조금 더 권한 설정을 어떻게 진행했는지, 어떤 명령어를 사용하셨는지 전체적인 스크린샷을 부탁드립니다! ㅎㅎ

**Reply by ggobsarikuna** [reply]
![image](https://user-images.githubusercontent.com/88932224/150081817-9e685205-f60b-480c-8b17-ebd068a70ff3.png)
이렇게 진행했습니다

**Reply by posquit0** [reply]
`chmod 400 ./fastcampus.pem` 명령어로 권한 설정 후 다시 시도해보시겠어요? ㅎㅎ

**Comment by ggobsarikuna** [comment]
똑같은 오류가 뜨네요... 혹시 wget 사용해서 구글드라이브에 ssh 키 올려놓고 다운받았는데 이 방법이 잘못된건가요?

**Reply by posquit0** [reply]
진행하신 환경이 윈도우 WSL 우분투환경이실까요~? 명령어를 실행하신 디렉토리 위치도 우분투 파일시스템인지, 윈도우 파일시스템인지 확인이 필요해보입니다! :)
chmod 400 명령어를 실행하면 소유자에 대한 r 읽기 권한이 허용되어야 하는데, 해당 내용에 대한 스샷 확인할 수 있을까요?

**Reply by posquit0** [reply]
https://serverfault.com/a/960647
위 링크의 커멘트도 확인해보시면 좋을 듯 합니다. 윈도우와 리눅스는 개행문자(엔터)를 다루는 인코딩 방법이 다릅니다. 파일을 여러번 옮겨다니면서 인코딩이 깨졌을 수 있습니다.

**Reply by ggobsarikuna** [reply]
드디어 해결했습니다. 감사합니다
